### PR TITLE
Surf heat balance arrays naming and initialization

### PR DESCRIPTION
**Pull request by xuanluo113**
Pull request overview
---------------------
The branch includes a lot of renaming and relocating surface heat balance arrays in `DataSurface`, `DataHeatBalance` and `DataHeatBalSurface`. Other minor changes are mostly in `SolarShading.cc`, including:
1. Reordering 3D arrays from (TimeStep, Hour, SurfNum) to (Hour, TimeStep, Surfnum) as the arrays are accessed by first Hour and then Timesteps (e.g. SurfSunlitFrac, SurfCosIncAng, SurfWinBackSurfaces).
2. Convert 3D array `SUNCOSTS` and `SurfSunCosHourly` to 2D of Vector3 arrays to avoid slicing and reconstructions.
3. Convert some implicit zero assignment of arrays to explicit loop assignment.
4. Reduce the scope of `Encl` only arrays from Zone to Enclosure.

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by xuanluo113** [comment]
@mjwitte This PR is ready for review as a follow-up of the last surface HB struct branch. However, I'm puzzled by the 0.064% performance done CI showed. Again, tested and profiled locally, the run time of this branch doe not show a consistent speed up and down comparing with the develop branch. The run time fluctuation between two runs is way larger than the CI performance diff and it's hard to tell.

**Comment by amirroth** [comment]
Are you expecting any performance changes here? This looks mostly like naming conventions. I would ignore any fluctuations in the ranges you are looking at.

File: doc/engineering-reference/src/climate-sky-and-solar-shading-calculations/shading-module.tex [hunk 1]
```diff
@@ -694,15 +694,15 @@ \subsubsection{Interior Solar Radiation Absorbed by Opaque Surfaces}\label{inter
...
                    &* [QD(OtherZoneNum) + ZoneIntGain(OtherZoneNum)\% QLTSW \\
                    &\quad + ZoneIntGain(OtherZoneNum)\% T\_QLTSW]
   \end{array}
 \end{equation}
 
 where
 
-\emph{FractDifShortZtoZ(OtherZoneNum,ZoneNum)} = ``diffuse solar exchange factor'' = fraction of short-wave radiation in \emph{OtherZoneNum} that is transmitted to \emph{ZoneNum}. This factor is calculated in subroutine ComputeDifSolExcZonesWIZWindows taking into account multiple reflection between zones. For example, for two zones means that some of the radiation transmitted from Zone1 to Zone2 is reflected back to Zone1, and some of this is in turn reflected back to Zone2, etc.
+\emph{ZoneFractDifShortZtoZ(OtherZoneNum,ZoneNum)} = ``diffuse solar exchange factor'' = fraction of short-wave radiation in \emph{OtherZoneNum} that is transmitted to \emph{ZoneNum}. This factor is calculated in subroutine ComputeDifSolExcZonesWIZWindows taking into account multiple reflection between zones. For example, for two zones means that some of the radiation transmitted from Zone1 to Zone2 is reflected back to Zone1, and some of this is in turn reflected back to Zone2, etc.
```
Comment (mjwitte) [inline doc/engineering-reference/src/climate-sky-and-solar-shading-calculations/shading-module.tex:705]: These may change with the addition of Spaces. And probably should be enclosure, but that's for the final frontier explorer to work out.

File: src/EnergyPlus/ChilledCeilingPanelSimple.cc [hunk 2]
```diff
@@ -1804,18 +1804,18 @@ Real64 SumHATsurf(EnergyPlusData &state, int const ZoneNum) // Zone number
 
             if (state.dataSurface->SurfWinFrameArea(SurfNum) > 0.0) {
                 // Window frame contribution
-                SumHATsurf += state.dataHeatBal->HConvIn(SurfNum) * state.dataSurface->SurfWinFrameArea(SurfNum) *
-                              (1.0 + state.dataSurface->SurfWinProjCorrFrIn(SurfNum)) * state.dataSurface->SurfWinFrameTempSurfIn(SurfNum);
+                SumHATsurf += state.dataHeatBalSurf->SurfHConvInt(SurfNum) * state.dataSurface->SurfWinFrameArea(SurfNum) *
+                              (1.0 + state.dataSurface->SurfWinProjCorrFrIn(SurfNum)) * state.dataSurface->SurfWinFramerTempIn(SurfNum);
```
Comment (mjwitte) [inline src/EnergyPlus/ChilledCeilingPanelSimple.cc]: `SurfWinFramerTempIn` should be `SurfWinFrameTempIn`.

File: src/EnergyPlus/DataBSDFWindow.hh [hunk 3]
```diff
@@ -442,17 +442,17 @@ struct BSDFWindowData : BaseGlobalStruct
     int TotThermalModels = 0;    // Number of thermal models
 
     // calculation
-    Array3D<Real64> SUNCOSTS = Array3D<Real64>(60, 24, 3);     // Timestep values of solar direction cosines
-    Array2D<Real64> BSDFTempMtrx;                              // Temporary matrix for holding axisymmetric input
-    EPVector<DataBSDFWindow::BSDFWindowGeomDescr> ComplexWind; // Window geometry structure: set in CalcPerSolarBeam/SolarShading
+    Array2D<Vector3<Real64>> SUNCOSTS = Array2D<Vector3<Real64>>(60, 24); // Timestep values of solar direction cosines
```
Comment (mjwitte) [inline src/EnergyPlus/DataBSDFWindow.hh:445]: Hooray! 1 less `Array3D`.

File: src/EnergyPlus/DataHeatBalSurface.hh [hunk 4]
```diff
@@ -81,146 +81,117 @@ struct HeatBalSurfData : BaseGlobalStruct
     std::vector<bool> Zone_has_mixed_HT_models; // True if any surfaces in zone use CondFD, HAMT, or Kiva
 
     // Integer Variables for the Heat Balance Simulation
-    Array1D_int SUMH; // From Old Bldctf.inc
+    Array1D_int SurfCurrNumHist; // From Old Bldctf.inc
```
Comment (mjwitte) [inline src/EnergyPlus/DataHeatBalSurface.hh:84]: That's an interesting name change - with much clearer meaning.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 5]
```diff
@@ -2764,25 +2761,26 @@ void InitSolarHeatGains(EnergyPlusData &state)
...
-                                                    state.dataHeatBal->EnclSolDBSSG(ZoneNum) + state.dataHeatBal->ZoneInitialDifSolReflW(ZoneNum);
+            state.dataHeatBal->EnclSolQD(enclosureNum) = state.dataHeatBal->EnclSolDB(enclosureNum) * state.dataEnvrn->BeamSolarRad +
+                                                         state.dataHeatBal->EnclSolDBSSG(enclosureNum) +
+                                                         state.dataHeatBal->ZoneInitialDifSolReflW(enclosureNum);
         }
 
         // Flux of diffuse solar in each zone
 
-        state.dataHeatBal->EnclSolQSDifSol = 0.0;
+        //        state.dataHeatBal->EnclSolQSDifSol = 0.0;
```
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: Delete?

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc [hunk 6]
```diff
@@ -3558,19 +3557,20 @@ void InitIntSolarDistribution(EnergyPlusData &state)
 
         for (int enclosureNum = 1; enclosureNum <= state.dataViewFactor->NumOfSolarEnclosures; ++enclosureNum) {
 
-            if (state.dataHeatBalSurf->RecDifShortFromZ(enclosureNum)) {
+            if (state.dataHeatBalSurf->ZoneRecDifShortFromZ(enclosureNum)) {
```
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: So, here it seems clear that `ZoneRecDifShortFromZ` is an enclosure variable, I think?

File: src/EnergyPlus/WindowComplexManager.cc [hunk 7]
```diff
@@ -958,35 +959,6 @@ namespace WindowComplexManager {
         }
     }
 
-    void DetermineMaxBackSurfaces(EnergyPlusData &state)
```
Comment (mjwitte) [inline src/EnergyPlus/WindowComplexManager.cc:961]: Interesting that this function was present but never used ever since BSDF was added.

**Review by mjwitte — COMMENTED** [review]
Looks good. A few comments.

**Comment by xuanluo113** [comment]
@mjwitte Thanks for the review! Comments addressed.

**Review by mjwitte — APPROVED** [review]
Looks good. Merging.

**Pull request merged**